### DISCUSSION TITLE
Modification in document.

### DISCUSSION THREAD

**Discussion by showbit01** [MatthiasValvekens/pyHanko#435] [Q&A]
I've signed a document and while checking for validation, it's showing valid in Ubuntu systems where as it's invalid in windows and Mac.
By looking into it deeply, I got to know that in Windows, it's showing that the document has been altered or modified where as for the same document it's not altered when opening in Ubuntu.
How could it be like this? I'm very confused.

**Comment by MatthiasValvekens** [comment]
The most likely answer is that the viewer you're using on Ubuntu (Okular, maybe?) doesn't validate incremental updates after signing. I'm actually not aware of even a single FOSS viewer supporting that kind of thing, because it's notoriously hard to do and there are no clear standards around. Assuming you're using Adobe Reader/Acrobat on Windows/macOS: it sometimes _appears_ to reject documents for benign reasons, but finding out exactly what makes it unhappy can be very hard even for subject matter experts (been there, done that).

Incidentally, PyHanko supports incremental update validation, but it is not 100% compatible with Acrobat (and it never will be, for that matter).
